ethereum-consensus/src/electra/block_processing.rs

```diff
@@ -107,7 +439,717 @@ pub fn process_execution_payload<
+            public_key,
+            withdrawal_credentials.clone(),
+            amount,
+            signature,
+            context,
+        )
+        .map(|_| true)?;
+        if is_compounding_withdrawal_credential(withdrawal_credentials) &&
+            has_eth1_withdrawal_credential(validator) &&
+            is_valid_deposit_signature
+        {
+            switch_to_compounding_validator(state, index, context)?;
+        }
+        return Ok(());
+    }
+
+    is_valid_deposit_signature(
+        public_key,
+        withdrawal_credentials.clone(),
+        amount,
+        signature,
+        context,
+    )?;
+    add_validator_to_registry(state, public_key.clone(), withdrawal_credentials.clone(), amount);
+    Ok(())
+}
+
+pub fn is_valid_deposit_signature(
+    public_key: &BlsPublicKey,
+    withdrawal_credentials: Bytes32,
+    amount: Gwei,
+    signature: &BlsSignature,
+    context: &Context,
+) -> Result<(), Error> {
+    let deposit_message =
+        DepositMessage { public_key: public_key.clone(), withdrawal_credentials, amount };
+    let domain = compute_domain(DomainType::Deposit, None, None, context)?;
+    let signing_root = compute_signing_root(&deposit_message, domain)?;
+    verify_signature(public_key, &signing_root[..], signature)
+}
+
+pub fn add_validator_to_registry<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const SYNC_COMMITTEE_SIZE: usize,
+    const BYTES_PER_LOGS_BLOOM: usize,
+    const MAX_EXTRA_DATA_BYTES: usize,
+    const PENDING_BALANCE_DEPOSITS_LIMIT: usize,
+    const PENDING_PARTIAL_WITHDRAWALS_LIMIT: usize,
+    const PENDING_CONSOLIDATIONS_LIMIT: usize,
+>(
+    state: &mut BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        SYNC_COMMITTEE_SIZE,
+        BYTES_PER_LOGS_BLOOM,
+        MAX_EXTRA_DATA_BYTES,
+        PENDING_BALANCE_DEPOSITS_LIMIT,
+        PENDING_PARTIAL_WITHDRAWALS_LIMIT,
+        PENDING_CONSOLIDATIONS_LIMIT,
+    >,
+    public_key: BlsPublicKey,
+    withdrawal_credentials: Bytes32,
+    amount: Gwei,
+) {
+    let index = state.validators.len();
+    state.validators.push(get_validator_from_deposit(public_key, withdrawal_credentials));
+    state.balances.push(0);
+    state.previous_epoch_participation.push(ParticipationFlags::default());
+    state.current_epoch_participation.push(ParticipationFlags::default());
+    state.inactivity_scores.push(0);
+    state.pending_balance_deposits.push(super::PendingBalanceDeposit { index, amount });
```

Comment:
generally prefer to import everything to a single identifier, and route imports from the `crate` root, rather than say `super` etc